sdk/core/core-http/lib/policies/deserializationPolicy.ts

```diff
@@ -262,8 +264,6 @@ export function deserializeResponseBody(
               const restError = new RestError(
                 `Error ${error} occurred in deserializing the responseBody - ${parsedResponse.bodyAsText}`
               );
-              restError.request = utils.stripRequest(parsedResponse.request);
-              restError.response = utils.stripResponse(parsedResponse);
```

Comment:
When there is an error in deserializing the response body, why shouldn't the resulting `RestError` have the `request` and `response` properties set?

Reply:
cc @xirzec

Reply:
These get set in the constructor now

Reply:
whoops that didn't happen here